app/views/products/_form.html.erb

```diff
@@ -0,0 +1,8 @@
+<%= form_for [ @vendor, @product ] do |f| %>
+  <%= f.label :name, "Name:" %>
+  <%= f.text_field :name, required: :true %>
+
+  <%= f.hidden_field :vendor_id, value: params[:vendor_id] %>
```

Comment:
Your `vendor_id` should be in the path you're submitting to, so making a hidden field usually isn't necessary. You might want to double check that your form is submitting to the right route if you don't have access to it. Also, this could be potentially dangerous. What's to stop someone from changing the hidden field's value to another vendor's id?